AndelaOSP/art-backend
# #159022596 Allow an asset to be assigned to a department

### What does this PR do?

- It allows an asset to be assigned to a department

### Description of task completed.

- write `AssetAssignee` model 
- Write post_save signals to save `AssetAssignee` when a `User` or a `Department` is saved
- fix tests to use new AssetAssignee as the `assigned_to`

### How should this be manually tested?

- visit the **allocations** endpoint with admin right and try to allocate an asset to a department

- Run tests to ensure nothing broke

### Screenshots
<img width="515" alt="screen shot 2018-08-14 at 01 57 53" src="https://user-images.githubusercontent.com/13068580/44062557-8dca0716-9f65-11e8-8135-511b0d7b5235.png">
